### PR TITLE
batch scheduler db changes

### PR DESCRIPTION
**Pull request by justinfisk**
* removed the status table and replaced with textfields with validator
* minor fix to the query api
* fixed (at least partially) rectify

### CODE REVIEW

**Comment by justinfisk** [comment]
Ah shoot. I accidentally committed my altered install.sh. I will fix.

File: gips/inventory/dbinv/migrations/0010_auto_20161215_1132.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                ('asset', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='dbinv.Asset')),
+            ],
+        ),
+        migrations.CreateModel(
+            name='ProductStatusChange',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('status', models.TextField(validators=[gips.inventory.dbinv.models.valid_status])),
+                ('time', models.DateTimeField(auto_now_add=True)),
+                ('product', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='dbinv.Asset')),
```
Comment (ags-tolson) [inline gips/inventory/dbinv/migrations/0010_auto_20161215_1132.py]: Should `to='dbinv.Asset'` be changed to `to='dbinv.Product'`?

Comment (ags-tolson) [inline gips/inventory/dbinv/migrations/0010_auto_20161215_1132.py]: Scratch that it got resolved in a further migration I believe.

File: gips/inventory/dbinv/models.py [hunk 2]
```diff
@@ -26,12 +27,26 @@ class Asset(models.Model):
...
-    status = models.ForeignKey(Status)         #
-
+    status = models.TextField(validators=[valid_status])
+    
     class Meta:
         # These four columns uniquely identify an asset file
         unique_together = ('driver', 'asset', 'tile', 'date')
 
+    # TODO: this is clunky and makes multiple round trips, and is repetitive
+    #       seems to beg to be a trigger
```
Comment (ags-tolson) [inline gips/inventory/dbinv/models.py]: agree that this doesn't look ideal but it looks like it works and it's a prototype so :+1: to this method

File: gips/inventory/dbinv/models.py [hunk 3]
```diff
@@ -47,13 +62,57 @@ class Product(models.Model):
...
+    """Dependencies of products on specific assets"""
+    product = models.ForeignKey(Product)
+    asset   = models.ForeignKey(Asset)
+    
+    class Meta:
+        unique_together = ('product', 'asset')
+
+        
+class AssetStatusChange(models.Model):
+    """Record of times product status updates"""
```
Comment (ags-tolson) [inline gips/inventory/dbinv/models.py]: Can you walk me through the use case for this model's uniqueness?  It looks like you can repeat an asset multiple times, but only once per valid status.  Is it expected that no asset can go 'backwards' to a state its already seen?  It seems like going from requested to in-progress to fail then back to in-progress might be a thing, but due to the way the `save` overload works, that last transition would raise an exception.

However, this is probably good enough for our rough and ready prototype.

File: gips/inventory/dbinv/models.py [hunk 4]
```diff
@@ -47,13 +62,57 @@ class Product(models.Model):
...
+    asset   = models.ForeignKey(Asset)
+    status  = models.TextField(validators=[valid_status])
+    time    = models.DateTimeField(auto_now_add=True)
+
+    class Meta:
+        unique_together = ('asset', 'status')
+
+        
+class ProductStatusChange(models.Model):
+    """Record of times product status updates"""
```
Comment (ags-tolson) [inline gips/inventory/dbinv/models.py]: Same question as for `AssetStatusChange`, just noting for code review purposes

**Pull request merged**